shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -32,4 +35,24 @@ public void itIgnoresNewHapticEventsOnOldAndroidPlatforms() {
     // SELECTION_CLICK haptic response is only available on "LOLLIPOP" (21) and later.
     platformPlugin.vibrateHapticFeedback(PlatformChannel.HapticFeedbackType.SELECTION_CLICK);
   }
+
+  @Test
+  public void platformPlugin_hasStrings() {
+    View fakeDecorView = mock(View.class);
+    Window fakeWindow = mock(Window.class);
+    when(fakeWindow.getDecorView()).thenReturn(fakeDecorView);
+    Activity fakeActivity = mock(Activity.class);
+    when(fakeActivity.getWindow()).thenReturn(fakeWindow);
+    PlatformChannel fakePlatformChannel = mock(PlatformChannel.class);
+    PlatformPlugin platformPlugin = new PlatformPlugin(fakeActivity, fakePlatformChannel);
+
+    ClipboardManager clipboardManager =
+        RuntimeEnvironment.application.getSystemService(ClipboardManager.class);
+    clipboardManager.setText("iamastring");
+
+    // TODO(justinmc): Can't do this because mPlatformMessageHandler is private.
+    // How can I send a message to the platform plugin?
+    //boolean hasStrings = platformPlugin.mPlatformMessageHandler.clipboardHasStrings();
+    //assertTrue(hasStrings);
```

Comment:
@gaaclarke You helped me test this same thing on iOS, do you know how I can do it on Android?  I'm sure there's a way for me to just send the Clipboard.hasStrings message, but I'm struggling to figure it out.

Reply:
I'd make a test against PlatformChannel that has a mock `platformMessageHandler` and you can call into `onMethodCall`.   As for testing PlatformPlugin, can't you just call `setClipboardData` on it then call `clipboardHasStrings` to test `clipboardHasStrings`?  Once you have those 2 tests written you'll have complete test coverage of your PR (in isolation as unit tests, not integration tests).